File: problem/35-Search-Insert-Position/main.py
"""
    Date:2021/10/28
"""
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        s = 0
        e = len(nums) - 1
        return self.search(nums, target, s, e)
    
    def search(self, nums: List[int], target: int, start , end):
        mid = (start + end) // 2
        
        if(start >= end):
            return start + (nums[start] < target)
        if(nums[mid] > target):
            return self.search(nums, target, start, mid-1)
        elif(nums[mid] < target):
            return self.search(nums, target, mid+1, end)
        else:
            return mid

File: problem/35-Search-Insert-Position/main.c
//Check the answer and write it...

int searchInsert(int* nums, int numsSize, int target){
    int i = 0, j = numsSize-1;
    int mid = -1;

    if(nums == NULL || numsSize == 0){
        return 0;
    }

    while(i <= j){
        mid = (i + j) / 2;

        if(nums[mid] == target){
            return mid;
        } else if(nums[mid] > target){
            j = mid - 1;
        } else if(nums[mid] < target){
            i = mid + 1;
        }
    }

    return i;
}
